### ISSUE TITLE
ansible-galaxy fails to install roles when specified version differs from repository's default branch

### ISSUE THREAD

**Issue by semifocused** [ansible/ansible#10477]
When we run ansible-galaxy specifying "version: some_branch_thats_not_the_repo_default_branch",

we see

```
- executing: git clone ssh://git@my.repo.com:/path/to/my/role.git rolename
- executing: git archive --prefix=rolename/ --output=/tmp/tmpaVd4UF.tar some_branch_thats_not_the_repo_default_branch
- command git archive --prefix=rolename/ --output=/tmp/tmpaVd4UF.tar some_branch_thats_not_the_repo_default_branch failed
  in directory /tmp/tmp0c9F7Y
- rolename was NOT installed successfully.
```

because the clone is using the default branch and has not pulled down the objects required to perform the export.  We have developed a fix that specifies the branch for the clone so the subsequent archive succeeds, which we will submit in a pull request.

**Comment by willthames** [comment]
@semifocused you can achieve the same results with tags.

It's actually nothing to do with not having the objects - it's the fact that version is a 'commit-ish' (i.e. you can use the first few hex digits of a commit id if you prefer) and shouldn't relate to a branch (which is not a specific point in time)

**Comment by fwisehc** [comment]
I'm experiencing the same issue in Ansible 2.0.1.0

Any version branch other than the default fails with the following message. 

[WARNING]: - hc_syslog was NOT installed successfully: - command git archive --prefix=hc_syslog/ --output=/tmp/tmpGbqok0.tar master failed in directory /tmp/tmppOIV7S (rc=128)

**Comment by willthames** [comment]
This is fixed with #10620 which will be released with 2.1.0

**Comment by jctanner** [comment]
This should have been resolved by https://github.com/ansible/ansible/pull/13203 which is in 2.1
